packages/web3/src/contract/script-simulator.ts

```diff
@@ -506,14 +527,24 @@ export class ScriptSimulator {
           break
         }
         case 'ApproveAlph': {
-          operandStack.popU256() // amount
-          operandStack.popAddress() // spender
+          const amount = operandStack.popU256() // amount
+          const spender = operandStack.popAddress() // spender
+          if (spender.kind.startsWith('Symbol')) {
+            approved.setUnknown() // The spender might be the caller
+          } else if (spender === callerAddress) {
+            approved.addApprovedAttoAlph(amount)
```

Comment:
Shall we set the approved to `unknown` as well if `callerAddress` isn't spender?

Reply:
We only show the approvals for call address in the return, so we don't need to care about other addresses.

Reply:
I am thinking about perhaps this would underestimate the total amount of approvals in corner cases, since we could have something like

```
foo.call{callerAddress!() -> 1 alph, alice -> 2 alph)
```

where alice is actually the same as callerAddress!(). Will this happen?

Reply:
If alice's address is unknown, then branch `spender.kind.startsWith('Symbol')` will be executed, the approved assets would be set to unknown.

If alice's address is known, then we will compare if the address is equal to caller address.